x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -74,6 +74,7 @@ public class SecurityServerTransportInterceptor implements TransportInterceptor
     static final Set<String> REMOTE_ACCESS_ACTION_ALLOWLIST;
     static {
         final Stream<String> actions = Stream.of(
+            TransportService.HANDSHAKE_ACTION_NAME,
```

Comment:
Put handshake under the allowlist. This change also serves as a test for the main change of this PR, i.e. the CCS REST test now sends handshake with the new model and it works because the connection is wrapped.

Reply:
Because the cluster connection is performed with system context. See AuthorizationUtils.shouldReplaceWithSytemContext (I am on mobile. So the name may not be accurate. So it works in this specific case. But you are right that for it to work in all cases, we want the new SeverFilter to also have the special logic to swap in a system user.

Reply:
Perfect, makes sense. Thanks for explaining 👍